crates/api/src/local_user/save_settings.rs

```diff
@@ -40,6 +42,17 @@ pub async fn save_user_settings(
   let bio = diesel_option_overwrite(
     process_markdown_opt(&data.bio, &slur_regex, &url_blocklist, &context).await?,
   );
+  if data.avatar.is_some() {
+    // Ignore errors because image may be stored externally.
+    if let Some(avatar) = &local_user_view.person.avatar {
+      let image = LocalImage::delete_by_url(&mut context.pool(), &avatar)
+        .await
+        .ok();
+      if let Some(image) = image {
+        delete_image_from_pictrs(&image.pictrs_alias, &image.pictrs_delete_token, &context).await?;
+      }
+    }
+  }
```

Comment:
This will go into a helper function.

Reply:
Thanks, I was just thinking of commenting abt that. Makes this a lot cleaner, even if it is only used once.